lib/cog/command/command_resolver.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Cog.Command.CommandResolver do
+
+  alias Cog.Repo
+  alias Cog.Queries.Command
+  alias Cog.Queries.Alias
+  alias Piper.Command.SemanticError
+
+  def find_bundle(name) when is_binary(name) do
+    if String.contains?(name, ":") do
+      :identity
+    else
+      case Repo.all(Command.bundle_for(name)) do
+        [bundle_name] ->
+          {:ok, bundle_name};
+        [] ->
+          case is_alias?(name) do
+            {true, alias_name} ->
+              {:ok, alias_name}
+            false ->
+              SemanticError.new(name, :no_command)
+          end
+        bundle_names ->
+          SemanticError.new(name, {:ambiguous_command, bundle_names})
+      end
+    end
+  end
+  def find_bundle(name) do
+    SemanticError.new("#{inspect name}", :no_command)
+  end
+
+  defp is_alias?("user:" <> name) do
```

Comment:
I think

`defp get_alias_type(String.t()) :: "site" | "user" | nil`

is a bit easier to understand.